sdcm/cluster.py

```diff
@@ -2995,18 +2985,8 @@ def node_setup(self, node, verbose=False, timeout=3600):
             except AssertionError:
                 disks = node.detect_disks(nvme=False)
             node.scylla_setup(disks)
-
-            seed_address_list = seed_address.split(',')
-            if node.ip_address not in seed_address_list:
-                wait.wait_for(func=lambda: self._seed_node_rebooted is True,
-                              step=30,
-                              text='Wait for seed node to be up after reboot')
-            node.restart()
-            node.wait_ssh_up()
-
-            if node.ip_address in seed_address_list:
-                self.log.info('Seed node is up after reboot')
-                self._seed_node_rebooted = True
+            # not sure why we need this reboot
+            node.reboot(hard=False, verify_ssh=True)
```

Comment:
The restart is add in initial support of GCE (commit  0738ca1483c9c8fa61c632afa45ffa45667740a4)

instance will stuck after raid setup for a while. We added 'sudo sync' and a instance restart to avoid this issue.

It's something 3 years ago, we need to check check if it works without the restart.
It should works, at least I didn't touch problem in manual testing.

Reply:
When do doing the reboot, we should be reboot the seed node first.

If we reboot non-seed nodes first, they will wait seed node. If seed node is setup a little bit late or reboot very late, other non-seed might timeout and won't start.

We can remote the reboot() and test if it works well.  Or continually reboot seed node first as past.

Reply:
I tested it and it works. Any-way, do you know why do we need reboot?

Reply:
In AWS implementation we restart Scylla Server service only.  I will try it instead of reboot

Reply:
> I tested it and it works. Any-way, do you know why do we need reboot?

`If we reboot non-seed nodes first, they will wait seed node. If seed node is setup a little bit late or reboot very late, other non-seed might timeout and won't start.`
This problem only occur sometimes.

Reply:
> In AWS implementation we restart Scylla Server service only. I will try it instead of reboot

Let's try.  I guess there is some kernel bug, that causes the system hang after raid setup without restart.

Reply:
@amoskong  BTW: After implementing new seed selection #1336  we first provision and setup seed nodes and then "regular" nodes.

Reply:
tested and it works without restarting the node.